hyperspy/signal.py

```diff
@@ -3808,16 +3805,46 @@ def diff(self, axis, order=1):
         (64,64,1024)
         >>> s.diff(-1).data.shape
         (64,64,1023)
-
         """
 
         s = self._deepcopy_with_new_data(
-            np.diff(self.data, order, axis))
-        axis = s.axes_manager._axes[axis]
-        axis.offset += (axis.scale / 2)
+            np.diff(self.data,
+                    n=order,
+                    axis=self.axes_manager[axis].index_in_array))
+        axis = s.axes_manager[axis]
+        axis.offset += (order * axis.scale / 2)
         s.get_dimensions_from_data()
         return s
 
+    def derivative(self, axis, order=1):
+        """Numerical derivative along the given axis.
+
+        Currently only the first order finete difference method is implemented.
+        Therefore, the
+        Parameters
+        ----------
+        axis : {int | string}
+           The axis can be specified using the index of the axis in
+           `axes_manager` or the axis name.
+        order: int
+            The order of the derivative
+
+        Returns
+        -------
+        def : Signal
+            The size of the
+
+        See also
+        --------
```

Comment:
I would have a look at the documentation again, to make sure that it states correct things and remove any unfinished sentences.